src/CoordinatorClusterClock.h

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2014-2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef RAMCLOUD_COORDINATORCLUSTERCLOCK_H
+#define RAMCLOUD_COORDINATORCLUSTERCLOCK_H
+
+#include "Common.h"
+#include "Context.h"
+#include "Cycles.h"
+#include "ExternalStorage.h"
+#include "WorkerTimer.h"
+
+namespace RAMCloud {
+
+/**
+ * The CoordinatorClusterClock, a module that runs on the coordinator, controls
+ * the progression of Cluster Time and maintains the following properties:
```

Comment:
Say what  Cluster Time is (hmm, maybe that's what you're trying to do below?), and give some idea of what it's used for.